client/app/main/student/student.controller.js

```diff
@@ -109,6 +109,50 @@ function StudentCtrl($scope, $stateParams, Intervention, Modal, Outreach,
       toastr.error(err);
     });
   };
+  $scope.toggleArchiveOutreach = function(outreach) {
+    var oldValue = outreach.archived;
+    var promise = Outreach.toggleArchive({
+      id: outreach._id
+    }, {
+      archived: !(outreach.archived)
+    }).$promise;
+    promise.then(function(outreach) {
+      $scope.$evalAsync(function() {
+        var outreaches = $scope.student.outreaches
+        outreaches.map(function(oldOutreach, i) {
+            if(oldOutreach._id === outreach._id) {
+              outreaches[i].archived = outreach.archived;
+            }
+        });
+      });
+      toastr.info(
+        'The ' + outreach.type +
+        ' outreach has been' +
+        (outreach.archived === true ? ' archived.' : ' unarchived.'));
+    }, function(err) {
+      outreach.archived = oldValue;
+      toastr.error(err);
+    });
+  };
```

Comment:
I took a look at this function and realized that the $evalAsync isn't needed. Angular's digest loop will pick up the changes.

```
  $scope.toggleArchiveOutreach = function(outreach) {
    Outreach.toggleArchive({
      id: outreach._id
    }, {
      archived: !outreach.archived
    }, function(toggledOutreach) {
      // Prefer not to use a variable name that is used in the surrounding scope.
      outreach.archived = toggledOutreach.archived;
      toastr.info(
        'The ' + outreach.type + ' outreach has been ' +
        (toggledOutreach.archived ? '' : 'un') + 'archived.'
    }, function(err) {
      // The value of archived is not changed before the server returns.
      // No op on the model on error. Just log it.
      console.log(err);
      toastr.error(err);
    });
  };
```

I'm thinking something like the code above without the comments. Since we don't need the promise we can just use the callbacks for the $resource for success and error. I'll make some comments on the original code but I think something like this might be good for this archive function.